doc/classes/ProjectSettings.xml

```diff
@@ -2673,6 +2673,8 @@
 		<member name="rendering/limits/spatial_indexer/update_iterations_per_frame" type="int" setter="" getter="" default="10">
 		</member>
 		<member name="rendering/limits/time/time_rollover_secs" type="float" setter="" getter="" default="3600">
+			Maximum time (in seconds) before the [code]TIME[/code] shader built-in variable rolls over. The [code]TIME[/code] variable increments by [code]delta[/code] each frame, and when it exceeds this value, it rolls over to [code]0.0[/code]. Since large floating-point values are less precise than small floating-point values, this should be set as low as possible to maximize the precision of the [code]TIME[/code] built-in variable in shaders. This is especially important on mobile platforms where precision in shaders is significantly reduced. However, if this is set too low, shader animations may appear to restart from the beginning while the project is running.
+			On desktop platforms, values below [code]4096[/code] are recommended, ideally below [code]2048[/code]. On mobile platforms, values below [code]64[/code] are recommended, ideally below [code]32[/code].
```

Comment:
It seems weird to me that we would recommend using values which are way below our default value. The default should be the recommended value, and if it needs to be so low for mobile, we would have a `.mobile` override.

What justifies such drastic recommendations?

Reply:
There is some useful context here: https://github.com/godotengine/godot/pull/59991 and https://github.com/godotengine/godot/pull/59990#issuecomment-1231596280

Also, for fun, you asked this same question already (https://github.com/godotengine/godot/pull/59990#discussion_r845491562)

The recommendation for 30 comes from that fact that for a 16-bit float, the precision is `0.03125` between 32 and 64. Its somewhat arbitrary, but anything higher starts to have really bad precision (`0.0625` for 64 to 128).

I'm not sure if a mobile override that is so drastic is a good idea since it will have a good chance to appearing like a shader is broken. 

lawnjelly, made a good recommendation on the [original PR for 3.x](https://github.com/godotengine/godot/pull/59991#issuecomment-1094612353):

> I noticed that Unity also has similar problems with time. If I remember right they try to solve it by storing some different divisors of time in uniforms (in a vec4), as well as some common sin values of time (in another vec4).
> 
> Imo though the only foolproof way to solve this mobile, say for water shaders, is for the client code to calculate a repeatable range (e.g a specific sin frequency) then e.g. send it as e.g. 0.0 to 1.0 in a custom uniform. Or multiple of these if there are multiple versions needed in a shader.
> 
> Sending time in a uniform and then doing these calculations in the shader seems like it will always introduce some kind of error on low precision hardware. Basically for low end mobile, TIME is probably best avoided I suspect.